cli/auth.go

```diff
@@ -28,9 +36,25 @@ var authCreateAdminToken = &cli.Command{
 
 		ctx := ReqContext(cctx)
 
+		if !cctx.IsSet("perm") {
+			return errors.New("--perm flag not set")
+		}
+
+		perm := cctx.String("perm")
+		idx := -1
+		for i, p := range api.AllPermissions {
+			if perm == p {
+				idx = i
+			}
+		}
+
+		if idx == -1 {
+			return fmt.Errorf("--perm flag has to be one of: %s", api.AllPermissions)
+		}
+
 		// TODO: Probably tell the user how powerful this token is
 
-		token, err := napi.AuthNew(ctx, api.AllPermissions)
+		token, err := napi.AuthNew(ctx, api.AllPermissions[:idx])
```

Comment:
hrm... weird, but fine. Probably want to remove that TODO comment, and then leave a comment saying why we're slicing here (admin perms give you all lesser perms, same for write, etc)